Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.cpp

```diff
@@ -515,10 +516,93 @@ std::optional<WebCore::FloatBoxExtent> ArgumentCoder<WebCore::FloatBoxExtent>::d
     };
 }
 
+enum class WebCore_TimingFunction_Subclass : uint8_t {
+    LinearTimingFunction,
+    CubicBezierTimingFunction,
+    StepsTimingFunction,
+    SpringTimingFunction
+}
+
+void ArgumentCoder<WebCore::TimingFunction>::encode(Encoder& encoder, const WebCore::TimingFunction& instance)
+{
+    if (auto* subclass = dynamicDowncast<WebCore::LinearTimingFunction>(instance.get())) {
+        encoder << WebCore_TimingFunction_Subclass::LinearTimingFunction;
+        encoder << *subclass;
+    }
+    if (auto* subclass = dynamicDowncast<WebCore::CubicBezierTimingFunction>(instance.get())) {
+        encoder << WebCore_TimingFunction_Subclass::CubicBezierTimingFunction;
+        encoder << *subclass;
+    }
+    if (auto* subclass = dynamicDowncast<WebCore::StepsTimingFunction>(instance.get())) {
+        encoder << WebCore_TimingFunction_Subclass::StepsTimingFunction;
+        encoder << *subclass;
+    }
+    if (auto* subclass = dynamicDowncast<WebCore::SpringTimingFunction>(instance.get())) {
+        encoder << WebCore_TimingFunction_Subclass::SpringTimingFunction;
+        encoder << *subclass;
+    }
+    ASSERT_NOT_REACHED();
+}
+
+std::optional<Ref<WebCore::TimingFunction>> ArgumentCoder<WebCore::TimingFunction>::decode(Decoder& decoder)
+{
+    std::optional<WebCore_TimingFunction_Subclass> type;
+    decoder >> type;
+    if (!type)
+        return std::nullopt;
+    if (type == WebCore_TimingFunction_Subclass::LinearTimingFunction) {
+        std::optional<WebCore::LinearTimingFunction> result;
+        decoder >> result;
+        if (!result)
+            return std::nullopt;
+        return WTFMove(*result);
+    }
+
+    if (type == WebCore_TimingFunction_Subclass::CubicBezierTimingFunction) {
+        std::optional<WebCore::CubicBezierTimingFunction> result;
+        decoder >> result;
+        if (!result)
+            return std::nullopt;
+        return WTFMove(*result);
+    }
+
+    if (type == WebCore_TimingFunction_Subclass::StepsTimingFunction) {
+        std::optional<WebCore::StepsTimingFunction> result;
+        decoder >> result;
+        if (!result)
+            return std::nullopt;
+        return WTFMove(*result);
+    }
+
+    if (type == WebCore_TimingFunction_Subclass::SpringTimingFunction) {
+        std::optional<WebCore::SpringTimingFunction> result;
+        decoder >> result;
+        if (!result)
+            return std::nullopt;
+        return WTFMove(*result);
+    }
+
+    ASSERT_NOT_REACHED();
+    return std::nullopt;
+}
+
 } // namespace IPC
 
 namespace WTF {
 
+template<> bool isValidEnum<WebCore_TimingFunction_Subclass, void>(uint8_t value)
```

Comment:
Does it compile successfully without this?  Strange if so, but if so then let's omit this.

Reply:
Yeah, it did look like the build was successful.